CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+cmake_minimum_required(VERSION 3.10)
+project(sg14 CXX)
+
+find_package(Threads REQUIRED)
+
+set(CMAKE_CXX_STANDARD 17)
+set(SG14_INCLUDE_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/SG14")
+set(SG14_TEST_SOURCE_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/SG14_test")
+
+# Output binary to predictable location.
+set(BINARY_OUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/bin)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${BINARY_OUT_DIR})
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${BINARY_OUT_DIR})
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${BINARY_OUT_DIR})
+
+foreach(OUTPUTCONFIG ${CMAKE_CONFIGURATION_TYPES})
+	string(TOUPPER ${OUTPUTCONFIG} OUTPUTCONFIG)
+	set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${BINARY_OUT_DIR})
+	set(CMAKE_LIBRARY_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${BINARY_OUT_DIR})
+	set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${BINARY_OUT_DIR})
+endforeach(OUTPUTCONFIG CMAKE_CONFIGURATION_TYPES)
+
+link_directories(${CMAKE_CURRENT_BINARY_DIR}/lib) # For future use with testing library.
+
+##
+# Project
+##
+file(GLOB_RECURSE INCLUDES "${SG14_INCLUDE_DIRECTORY}/*.hpp" "${SG14_INCLUDE_DIRECTORY}/*.h" "${SG14_INCLUDE_DIRECTORY}/*.tpp")
```

Comment:
What is `*.tpp`? I've never seen a project with `.tpp` files. (But I have seen `.hh`, `.H`, and `.ipp`. But not in the SG14 repo, so those extensions shouldn't be in our cmake file either, I wouldn't think.)

Reply:
I will remove it. tpp is often used to separate your template declarations from implementations (I'm guessing it is the same as ipp). At the bottom of your `header.hpp`, you `#include header.tpp`. It is used as a convenience to separate really large headers.